internal/connect/system.go

```diff
@@ -150,22 +151,22 @@ func ReadOnlyFilesystem(root string) error {
 		return fmt.Errorf("Checking whether %v is mounted read-only failed: %v", path, err)
 	}
 
-	// Just like zypper, we will assume that a BTRFS file system with the
-	// `transactional-update` binary installed is a transactional server.
-	_, err := os.Stat("/usr/sbin/transactional-update")
-	if statfs.Type == BTRFS_SUPER_MAGIC && err == nil {
-		// The user did not use the 'root' flag from the CLI: this is not
-		// `transactional-update` calling SUSEConnect but rather a user
-		// directly. This is not allowed.
-		if root == "" {
-			return errors.New("This is a transactional-server, please use `transactional-update register` to manage your product activations")
-		}
-	}
-
-	// The root is read only, we cannot write in there.
 	if (statfs.Flags & ST_RDONLY) == ST_RDONLY {
+		// Just like zypper, we will assume that a BTRFS file system with the
+		// `transactional-update` binary installed is a transactional server.
+		_, err := os.Stat("/usr/sbin/transactional-update")
+		if statfs.Type == BTRFS_SUPER_MAGIC && err == nil {
+			// The user did not use the 'root' flag from the CLI: this is not
+			// `transactional-update` calling SUSEConnect but rather a user
+			// directly which is dicouraged.
+			if root == "" {
+				return errors.New("This is a transactional-server, please use `transactional-update register` to manage your product activations")
```

Comment:
Unrelated to the code change, but where is the term "transactional-server" defined? I've tried to find it, but to not avail. Is there another, more common term that users would understand better?

Reply:
Probably doesn't answer your question, but: It used to appear in the ISO Installer

Reply:
@lagartoflojo I asked Ignaz about this. The official term is `transactional system` as the install pattern in SLE. I change the naming here now

Reply:
Thank you Felix!